### PR TITLE
AWS EC2 VPC Gather Information section added

### PR DESCRIPTION
**Pull request by navybk**
Since the aws ec2 cli delivers other objects than the ansible ec2 core plugins, there should be a possibility to read the subnets of an existing vpc.

With this pull request all you need is to define "state" as "getinfo" and provide the vpc-id.

### CODE REVIEW

**Comment by bennojoy** [comment]
@navybk  i think should we also check if the user has not entered a valid vpc_id , then respond with message that vpc_id does not exist

**Comment by bcoca** [comment]
I would think this belongs in a fact gathering module

**Comment by navybk** [comment]
Since the information are gathered within the module anyway, it would make no sense to create an other module with partly _identical_ code.
And in the end when working with the module you expect to get information in response, if a vpc with the id exists

**Comment by gregdek** [comment]
@carsongee Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please give it a +1; at that point, we will add the “priority_reviewed” flag to this PR, which will move it to the top of our queue for potential inclusion.

**Comment by gregdek** [comment]
@carsongee ping -- are you available to review this PR?

File: cloud/amazon/ec2_vpc.py [hunk 1]
```diff
@@ -656,6 +656,28 @@ def terminate_vpc(module, vpc_conn, vpc_id=None, cidr=None):
 
     return (changed, vpc_dict, terminated_vpc_id)
 
+def get_vpc (module, vpc_conn, vpc_id):
+
+    if vpc_id == None and cidr == None:
```
Comment (carsongee) [inline cloud/amazon/ec2_vpc.py:661]: This conditional isn't needed, since `find_vpc` will do this check.

File: cloud/amazon/ec2_vpc.py [hunk 2]
```diff
@@ -656,6 +656,28 @@ def terminate_vpc(module, vpc_conn, vpc_id=None, cidr=None):
 
     return (changed, vpc_dict, terminated_vpc_id)
 
+def get_vpc (module, vpc_conn, vpc_id):
```
Comment (carsongee) [inline cloud/amazon/ec2_vpc.py:659]: I think you should pass in cidr in case people want to use the cidr + tags feature of `find_vpc`.

**Comment by carsongee** [comment]
I'm fine with this going in after the suggested fixes, but I also think this would be really cool to add into the ec2.py inventory script in addition to this.

**Comment by carsongee** [comment]
Also, sorry for the delay, I didn't see this when I first got pinged a few months ago :blush:

**Comment by gregdek** [comment]
Hi @navybk -- going through cases where our bot missed old PRs, and seeing this one. Looks like @carsongee asked for some revisions, so putting this PR into needs_revision. Please make the suggested revisions to your PR, and when you've done so, please comment with "ready_for_review" so we can put it back into community_review. Thanks!

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@navybk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**